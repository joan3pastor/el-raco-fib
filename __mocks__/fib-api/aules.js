export const quadri_actual = {
    "id": "2020Q1",
    "url": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/",
    "actual": "S",
    "actual_horaris": "S",
    "classes": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/classes/",
    "examens": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/examens/",
    "assignatures": "https://api.fib.upc.edu/v2/quadrimestres/2020Q1/assignatures/"
};

export const aules = {
    "count": 51,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": "A1S101"
        },
        {
            "id": "A3204"
        },
        {
            "id": "A4002"
        },
        {
            "id": "A5001"
        },
        {
            "id": "A5002"
        },
        {
            "id": "A5101"
        },
        {
            "id": "A5102"
        },
        {
            "id": "A5103"
        },
        {
            "id": "A5104"
        },
        {
            "id": "A5105"
        },
        {
            "id": "A5106"
        },
        {
            "id": "A5201"
        },
        {
            "id": "A5202"
        },
        {
            "id": "A5203"
        },
        {
            "id": "A5204"
        },
        {
            "id": "A5205"
        },
        {
            "id": "A5206"
        },
        {
            "id": "A5E01"
        },
        {
            "id": "A5E02"
        },
        {
            "id": "A5S108"
        },
        {
            "id": "A5S109"
        },
        {
            "id": "A5S112"
        },
        {
            "id": "A5S113"
        },
        {
            "id": "A6001"
        },
        {
            "id": "A6002"
        },
        {
            "id": "A6101"
        },
        {
            "id": "A6102"
        },
        {
            "id": "A6103"
        },
        {
            "id": "A6104"
        },
        {
            "id": "A6105"
        },
        {
            "id": "A6106"
        },
        {
            "id": "A6201"
        },
        {
            "id": "A6202"
        },
        {
            "id": "A6203"
        },
        {
            "id": "A6204"
        },
        {
            "id": "A6205"
        },
        {
            "id": "A6206"
        },
        {
            "id": "A6E01"
        },
        {
            "id": "A6E02"
        },
        {
            "id": "B5S201"
        },
        {
            "id": "B5S202"
        },
        {
            "id": "C5S202"
        },
        {
            "id": "C5S203"
        },
        {
            "id": "C6S301"
        },
        {
            "id": "C6S302"
        },
        {
            "id": "C6S303"
        },
        {
            "id": "C6S306"
        },
        {
            "id": "C6S308"
        },
        {
            "id": "C6S309"
        },
        {
            "id": "D6003"
        },
        {
            "id": "Online"
        }
    ]
};

