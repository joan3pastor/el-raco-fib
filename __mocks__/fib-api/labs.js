export const labImages = {
    "imatges": {
        "A5": "https://api.fib.upc.edu/v2/laboratoris/imatges/A5.png",
        "C6": "https://api.fib.upc.edu/v2/laboratoris/imatges/C6.png",
        "B5": "https://api.fib.upc.edu/v2/laboratoris/imatges/B5.png"
    }
}
