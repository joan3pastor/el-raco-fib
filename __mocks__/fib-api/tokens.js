
export const token_buit = { }

export const token = {
    token: "token_mock",
    refresh_token: "refresh_token_mock",
    expire_date: "3593803220670"
}

export const token_valid = {
    token: "token_valid_original",
    refresh_token: "refresh_token_mock",
    expire_date: "2540757600000"
}

export const token_caducat = {
    token: "token_caducat_original",
    refresh_token: "refresh_token_mock",
    expire_date: "1530914400000"
}

export const token_nou = {
    access_token: "token_valid_nou",
    refresh_token: "refresh_token_mock",
    expire_date: "5555555500000"
}