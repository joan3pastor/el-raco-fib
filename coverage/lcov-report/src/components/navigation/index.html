<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/navigation</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/components/navigation
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.01% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>337/432</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>191/221</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.61% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>77/141</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>281/371</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Assignatures.js"><a href="Assignatures.js.html">Assignatures.js</a></td>
	<td data-value="82.35" class="pic high"><div class="chart"><div class="cover-fill" style="width: 82%;"></div><div class="cover-empty" style="width:18%;"></div></div></td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="22" class="abs high">20/22</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	</tr>

<tr>
	<td class="file high" data-value="Avisos.js"><a href="Avisos.js.html">Avisos.js</a></td>
	<td data-value="81.98" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.98" class="pct high">81.98%</td>
	<td data-value="172" class="abs high">141/172</td>
	<td data-value="91.45" class="pct high">91.45%</td>
	<td data-value="117" class="abs high">107/117</td>
	<td data-value="54.29" class="pct medium">54.29%</td>
	<td data-value="35" class="abs medium">19/35</td>
	<td data-value="77.17" class="pct medium">77.17%</td>
	<td data-value="127" class="abs medium">98/127</td>
	</tr>

<tr>
	<td class="file low" data-value="Drawer.js"><a href="Drawer.js.html">Drawer.js</a></td>
	<td data-value="47.37" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	</tr>

<tr>
	<td class="file high" data-value="Events.js"><a href="Events.js.html">Events.js</a></td>
	<td data-value="81.03" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.03" class="pct high">81.03%</td>
	<td data-value="58" class="abs high">47/58</td>
	<td data-value="67.65" class="pct medium">67.65%</td>
	<td data-value="34" class="abs medium">23/34</td>
	<td data-value="86.67" class="pct high">86.67%</td>
	<td data-value="15" class="abs high">13/15</td>
	<td data-value="78.43" class="pct medium">78.43%</td>
	<td data-value="51" class="abs medium">40/51</td>
	</tr>

<tr>
	<td class="file low" data-value="ExternalLinks.js"><a href="ExternalLinks.js.html">ExternalLinks.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Feedback.js"><a href="Feedback.js.html">Feedback.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file high" data-value="Horari.js"><a href="Horari.js.html">Horari.js</a></td>
	<td data-value="89.86" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.86" class="pct high">89.86%</td>
	<td data-value="69" class="abs high">62/69</td>
	<td data-value="86.11" class="pct high">86.11%</td>
	<td data-value="36" class="abs high">31/36</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="66" class="abs high">60/66</td>
	</tr>

<tr>
	<td class="file high" data-value="Labs.js"><a href="Labs.js.html">Labs.js</a></td>
	<td data-value="81.25" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="16" class="abs high">13/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="Noticies.js"><a href="Noticies.js.html">Noticies.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="42" class="abs medium">30/42</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	</tr>

<tr>
	<td class="file high" data-value="SalesBibl.js"><a href="SalesBibl.js.html">SalesBibl.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jul 17 2020 06:12:00 GMT+0200 (GMT+02:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
