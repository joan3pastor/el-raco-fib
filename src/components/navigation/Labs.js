import React, { Component } from 'react';
import { Text, View, Image, Dimensions, ScrollView, Button } from 'react-native';
import { connect } from 'react-redux';
import _ from 'lodash';

import { getToken } from '../../actions';
import updateToken from '../../actions/updateToken';

export class Labs extends Component {

  constructor(props) {
    super(props);
    this.state = {update:false};
  }

  componentDidMount() {
    this.refresh();
  }

  async refresh() {
    updateToken.bind(this)().then(() => this.setState({update: JSON.stringify(Date.now())}));
  }

  renderLabs() {
    var windowWidth = Dimensions.get('window').width;
    var labs = [];
    var reload_date = JSON.stringify(Date.now())
    if (!_.isUndefined(this.props.testsnapshot)) reload_date = "1";
    _.forOwn(this.props.imglabs.imatges, (lab, key) => {
      labs.push(
        <View key={key} style={{backgroundColor:'#fafafa'}}>
          <Text style={{alignSelf:'center', fontSize:17, padding:5, marginTop:15, fontWeight:'bold', color:'#424848'}}>{key}</Text>
          <Image 
            source={{uri: `${lab}?reload=${reload_date}`, headers: {
              'Authorization': `Bearer ${this.props.token.token}`, 
              'Accept-Language': 'ca'
            }}}
            style={{width:windowWidth, height:250, resizeMode:'stretch'}}
          />
        </View>
      );
    });
    return labs;
  }

  render() {
    return (
      <ScrollView style={{ flex: 1, backgroundColor:'#fafafa' }}>
        <Button
          onPress={() => this.refresh()}
          title={this.props.lang.bttnActualitzar}
          color="#1E90FF"
        />
        {this.renderLabs()}
      </ScrollView>
    );
  }
}

function mapStateToProps(state) {
  return { 
    imglabs: state.imglabs,
    token: state.token,
    lang: state.lang.labs,
  };
}

export default connect(mapStateToProps, { getToken })(Labs);
