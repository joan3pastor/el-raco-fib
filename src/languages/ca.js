export const ca = {
    code: "ca",
    bottomTabs: { 
        horari: "HORARI",
        avisos: "AVISOS",
        noticies: "NOTÍCIES",
        assig: "ASSIG.",
    },
    titleViews: { 
        events: "Esdeveniments", 
        notes: "Gestor de Notes", 
        labs: "Disponibilitat Labs", 
        bibl: "Sales Biblioteca BRGF", 
        aules: "Cerca Aules Lliures", 
        places: "Seguiment Places Matr.", 
        mobilitat: "Programes de Mobilitat", 
        lectures: "Lectures TFG i TFM", 
        links: " Enllaços d'Interès ", 
        feedback: "Feedback & Bugs", 
        settings: "Configuració", 
    },
    drawer: { 
        events: "Esdeveniments",
        notes: "Gestor de Notes",
        labs: "Disponibilitat Labs",
        bibl: "Sales Bibl. BRGF",
        aules: "Cerca Aules Lliures",
        places: "Places Matrícula",
        mobilitat: "Programes de Mobilitat",
        lectures: "Lectures TFG i TFM",
        links: "Enllaços d'Interès",
        feedback: "Feedback & Bugs",
        settings: "Configuració",
        logout: "Log Out",
    },
    horari: { 
        grup: "Grup",
        aula: "Aula",
        diesSetmana: ["dl.", "dt.", "dc.", "dj.", "dv."],
    },
    avisos: {
        msgEmpty: "No tens avisos.",
    }, 
    noticies: {}, 
    assig: { 
        grup: "Grup",
        credits: "Credits",
        semestre: "Semestre",
        professors: "Professors",
        avaluacio: "Mètode d'avaluació",
        metodologia: "Metodologia docent",
        loadingProfessors: "Carregant professors...",
        loadingAvaluacio: "Carregant avaluació...",
        loadingMetodologia: "Carregant metodologia...",
        responsable: "Responsable",
        msgEmpty: "No estàs cursant cap assignatura.",
    },
    events: { 
        footer: "Només mostra exàmens en hores no lectives i dies festius fins a 3 mesos endevant.",
    },
    notes: { 
        mitja: "Nota mitja",
        final: "Nota total",
        progres: "Progrés",
        bttnAfegir: "AFEGIR ENTRADA",
        bttnCancellar: "CANCEL·LAR",
        bttnGuardar: "GUARDAR",
        aeNom: "Nom acte avaluatiu",
        aePes: "Pes (0 a 1)",
        aeNota: "Nota",
        textNota: "Nota",
        titolEliminar: "Atenció!",
        textEliminar: "Segur que vols eliminar l'entrada?  ",
        textSi: "SI",
        textNo: "NO",
        empty: "No estàs cursant cap assignatura.",
    },
    labs: { 
        bttnActualitzar: "ACTUALITZAR",
    },
    bibl: {}, 
    aules: { 
        pickerDiaNom: "Dia",
        pickerIniciNom: "Inici",
        pickerDuradaNom: "Durada",
        pickerTipusNom: "Tipus",
        pickerDia: ["Avui", "dl.", "dt.", "dc.", "dj.", "dv."],
        pickerDiaFormat: [{value:"Avui"}, {value:"dl."}, {value:"dt."}, {value:"dc."}, {value:"dj."}, {value:"dv."}],
        pickerTipus: ["Tots", "AULA", "LAB"],
        pickerTipusFormat: [{value:"Tots"}, {value:"AULA"}, {value:"LAB"}],
        msgSelecDia: "Selecciona un dia.",
        msgSelecInici: "Selecciona una hora d'inici.",
        msgSelecDurada: "Selecciona una durada mínima.",
        sectionAules: "Aules lliures",
        sectionLabs: "Labs lliures",
        aulaMsgLinia1: "Lliure fins",
        aulaMsgLinia2: "les",
    },
    places: { 
        bttnAfegir: "Afegir Assig.",
        bttnActualitzar: "Actualitzar",
        msgActualitzant: "Actualitzant en",
        msgBuscarAssig: "Buscar Assignatura",
        msgGrup: "Grup",
        cursos: {
            "GRAU": "Grau en Enginyeria Informàtica",
            "GCED": "Grau en Ciència i Enginyeria de Dades",
            "MEI": "Màster en Enginyeria Informàtica",
            "MIRI": "Màster en Innovació i Recerca en Informàtica",
            "MSEC": "Màster en Formació ESO i Batxillerat",
            "MAI": "Màster en Intel·ligència Artificial",
        },
        titolEliminar: "Atenció!",
        textEliminar: "Segur que vols eliminar l'entrada?  ",
        textSi: "SI",
        textNo: "NO",
    },
    mobilitat: { 
        pickerPais: "País",
        pickerPrograma: "Programa",
        pickerElemTots: "Tots",
        modalTitol: "Enllaços d'Interès",
        modalLink1: "Informació General",
        modalLink2: "Sessions Informatives",
        modalLink3: "Calendari",
        modalLink4: "Beques",
        bttnWeb: "Pàgina Web",
        bttnMapa: "Obrir Mapa",
        bttnPDF: "PDF Info",
        msgPais: "País",
        msgPrograma: "Programa",
        msgVigent: "Vigent fins",
    },
    lectures: { 
        msgBuit: "No hi ha lectures programades.",
        pickerPla: "Pla",
        pickerElemTots: "Tots",
        director: "Director",
        estudiant: "Estudiant",
        dia: "Dia",
        lloc: "Lloc",
        modalitat: "Modalitat",
        pla: "Pla",
    },
    links: { 
        title1: "La Facultat",
        title2: "Guies",
        title3: "Programari i Serveis per a l'Estudiant",
        A1E1: "DEFIB (Delegació d'Estudiants de la FIB)",
        A1E1Comentari: "   - Bona secció de FAQ per a nous estudiants.",
        A1E2: "Associacions",
        A2E1: "Manual de connexió a Eduroam (Wi-Fi)",
        A2E2: "Configuració del correu d'estudiant",
        A2E3: "Servei d'impressió",
        A2E4: "Configuració UPClink (VPN per accedir als servidors de la FIB des de fora la facultat)",
        A3E1: "Acords de distribució de programari UPC (Microsoft Imagine, Autocad, Matlab...)",
        A3E2: "VMware Academic Program (Ofert per la FIB)",
        A3E3: "Github Student Developer Pack (Github private projects, ASW Educate, Heroku Free Dynos, etc.)",
        A3E4: "Microsoft Azure for Students",
        A3E5: "Microsoft Office for Students + 1 TB OneDrive",
        A3E6: "Google Cloud Platform Education Credits",
        msgManteniment: "Si algun enllaç deixa de funcionar envieu-me un correu des de l'apartat \"Feedback & Bugs\".",
    },
    feedback: { 
        text: "<style>p{text-align:justify}</style><p>Siéntete libre de enviarme un correo si algo no se comporta como esperabas o si echas en falta alguna funcionalidad. Adjunta capturas de pantalla de todo lo posible.</p><p>Trataré de solucionar errores críticos. Todo lo demás dependerá de mi disponibilidad.</p>",
        bttnSendMail: "Envíame un correo!",
        bttnValora: "Valora la App",
        bttnGithub: "Colabora!",
        footer: "Creada por Joan Marc Pastor con la API del Racó",
    },
    settings: { 
        A1Title: "Notificacions avisos",
        A1Bttn1: "Activades",
        A1Bttn2: "Desactivades",
        A1Nota1: "Es buscaran nous avisos cada 20 minuts.",
        A1Nota2: "*Pot no funcionar en \"Power Saving Mode\".",
        A2Title: "Mida de la font dels avisos",
        A2text: "Text de prova...",
        A3Title: "Estil Horari",
        A3Bttn1: "Complet",
        A3Bttn2: "Minimalista",
        A4Title: "Idioma",
        A4Placeholder: "Selecciona idioma",
        dies: ["dl.", "dt.", "dc.", "dj.", "dv."],
        pickerDies: [{value:"dl."}, {value:"dt."}, {value:"dc."}, {value:"dj."}, {value:"dv."}],
        A5Title: "Modificar Horari",
        A5Bttn1: "Afegir Classe",
        A5Bttn2: "Eliminar Classe",
        A5Bttn3: "Reset",
        A5txt1: "Classes afegides",
        A5txt2: "Classes eliminades",
        A5ModalTitle: "Afegir classe",
        A5ModalP1: "Codi assig.",
        A5ModalP2: "Aula",
        A5ModalP3: "Dia",
        A5ModalP4: "Inici",
        A5ModalP5: "Durada",
        A5ModalP6: "Grup",
        A5ModalP7: "Tipus",
        A5ModalBttn1: "CANCEL·LAR",
        A5ModalBttn2: "GUARDAR",
        A5Modal2Title: "Eliminar classes",
        A5Modal2txt1: "Classes oficials",
        A5Modal2txt2: "Classes afegides",
        A5Alert1: "Introdueix el codi de l'assignatura.",
        A5Alert2: "Introduce la aula de la clase.",
        A5Alert3: "Introdueix la aula de la classe.",
        A5Alert4: "Introdueix l'hora d'inici de la classe.",
        A5Alert5: "Introdueix la durada de la classe.",
        A5Alert6: "Introdueix el grup.",
        A5Alert7: "Introdueix el tipus de la classe. \n\nT = Teoria\nL = Lab\nP = Problemes",
    },    
}
