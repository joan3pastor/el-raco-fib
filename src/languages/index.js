import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};