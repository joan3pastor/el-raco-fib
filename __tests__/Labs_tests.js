import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import AxiosMockAdapter from 'axios-mock-adapter';
import _ from 'lodash';

// LANG
import {ca} from "../src/languages/ca";
const lang = ca.labs;

import {Labs} from "../src/components/navigation/Labs";
import {labImages} from "../__mocks__/fib-api/labs";
import {token} from "../__mocks__/fib-api/tokens";

Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {};

beforeEach( () => {})

const wrapper = shallow(
  <Labs token={{token:undefined}} imglabs={labImages} testsnapshot={true} store={mockStore(initialState)} lang={lang} />
);

const instance = wrapper.instance();

describe('Testing Labs component', () => {
  it('renders as expected when token not yet provided', () => {
    wrapper.setProps({lang:lang});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected when token provided', () => {
    wrapper.setProps({token: token, lang:lang});
    expect(wrapper).toMatchSnapshot();
  });
});
