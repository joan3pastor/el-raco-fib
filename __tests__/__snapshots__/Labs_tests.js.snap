// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Labs component renders as expected when token not yet provided 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <Button
    color="#1E90FF"
    onPress={[Function]}
    title="ACTUALITZAR"
  />
  <View
    key="A5"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      A5
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer undefined",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/A5.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
  <View
    key="C6"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      C6
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer undefined",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/C6.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
  <View
    key="B5"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      B5
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer undefined",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/B5.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
</ScrollViewMock>
`;

exports[`Testing Labs component renders as expected when token provided 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <Button
    color="#1E90FF"
    onPress={[Function]}
    title="ACTUALITZAR"
  />
  <View
    key="A5"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      A5
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer token_mock",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/A5.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
  <View
    key="C6"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      C6
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer token_mock",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/C6.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
  <View
    key="B5"
    style={
      Object {
        "backgroundColor": "#fafafa",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#424848",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginTop": 15,
          "padding": 5,
        }
      }
    >
      B5
    </Text>
    <Image
      source={
        Object {
          "headers": Object {
            "Accept-Language": "ca",
            "Authorization": "Bearer token_mock",
          },
          "uri": "https://api.fib.upc.edu/v2/laboratoris/imatges/B5.png?reload=1",
        }
      }
      style={
        Object {
          "height": 250,
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
  </View>
</ScrollViewMock>
`;
